### PR TITLE
TauP introduction and install

### PR DESCRIPTION
**Pull request by core-man**
**Preview**: https://software-git-taup-tutorial.seismo-learn.vercel.app

1. Split TauP tutorial into three parts: brief intro; install; usage
2. Finish the draft of intro and install
3. Leave usage TODO﻿

see also #1

### CODE REVIEW

**Comment by core-man** [comment]
@seisman, please comment on the format as I am not very familiar with good rst format writing.

**Comment by seisman** [comment]
Looks good. Shall we split the content into several smaller files?

**Comment by core-man** [comment]
yes, it is a little long for a single file. maybe it is better to split into several files in the next commit.

We may also make a rule about the best length and hard limit one of a simple file, which may be included in the template #24 .

File: source/taup.rst [hunk 1]
```diff
@@ -7,14 +7,125 @@ TauP 是一款常用于计算地震波走时和路径的 Java 程序。
...
+    $ sudo yum install java
+
+    # 在 Ubuntu 上
+    $ sudo apt update
+    $ sudo apt install default-jre
+    $ sudo apt install default-jdk
+    $ sudo apt upgrade
+
+    # 在 macOS 上
+    $ brew cask install java
```
Comment (seisman) [inline source/taup.rst:72]:
```suggestion
    $ brew install openjdk
```

File: source/taup.rst [hunk 2]
```diff
@@ -7,14 +7,125 @@ TauP 是一款常用于计算地震波走时和路径的 Java 程序。
...
+否则需要先安装 Java 运行环境::
+
+    # 在 CentOS 7 上
+    $ sudo yum install java
+
+    # 在 Ubuntu 上
+    $ sudo apt update
+    $ sudo apt install default-jre
+    $ sudo apt install default-jdk
+    $ sudo apt upgrade
```
Comment (seisman) [inline source/taup.rst:69]: No need to run `apt upgrade`
```suggestion
```

File: source/taup.rst [hunk 3]
```diff
@@ -7,14 +7,125 @@ TauP 是一款常用于计算地震波走时和路径的 Java 程序。
...
+    $ tar -xvf TauP-2.4.5.tgz
+
+4. 安装::
+
+    $ sudo mv TauP-2.4.5 /opt
+
+5. 修改环境变量::
+
+    $ echo 'export TAUPHOME=/opt/TauP-2.4.5' >> ~/.bashrc
+    $ echo 'export PATH=${TAUPHOME}/bin:${PATH}' >> ~/.bashrc
```
Comment (seisman) [inline source/taup.rst:92]: OK to merge these two lines:
```suggestion
    $ echo 'export PATH=/opt/TauP-2.4.5/bin:${PATH}' >> ~/.bashrc
```

File: source/taup.rst [hunk 4]
```diff
@@ -7,14 +7,125 @@ TauP 是一款常用于计算地震波走时和路径的 Java 程序。
...
+
+否则需要先安装 Java 运行环境::
+
+    # 在 CentOS 7 上
+    $ sudo yum install java
+
+    # 在 Ubuntu 上
+    $ sudo apt update
+    $ sudo apt install default-jre
+    $ sudo apt install default-jdk
```
Comment (seisman) [inline source/taup.rst:68]: Merge these two lines, although I can't confirm the commands are correct.
```suggestion
    $ sudo apt install default-jre default-jdk
```

**Comment by seisman** [comment]
Perhaps we should always use lower case names for directories?

https://software-git-taup-tutorial.seismo-learn.vercel.app/TauP/install/ looks a little weird.

File: .gitignore [hunk 5]
```diff
@@ -1 +1,3 @@
 build
+_build
```
Comment (seisman) [inline .gitignore:2]: Please make the irrelevant changes in a separate PR instead.

File: source/taup/intro.rst [hunk 6]
```diff
@@ -0,0 +1,40 @@
+简介
```
Comment (seisman) [inline source/taup/intro.rst:1]: Please rewrite the software metadata information following [the template](https://github.com/seismo-learn/software/pull/27).

File: .gitignore (hunk 5 above)
Comment (core-man) [reply .gitignore:2]: Okay. You mean remove _build and .vscode in .gitignore and submit a new PR now?

Comment (seisman) [reply .gitignore:2]: You don't have to remove them from .gitignore. You can just submit a new PR. When that PR is merged into master, it should be straightforward to merge master into this branch.

Comment (seisman) [reply .gitignore:2]: Now you can click the "Update branch" button to merge the master bracnch

Comment (core-man) [reply .gitignore]: 真滴方便啊~

**Comment by seisman** [comment]
**Preview**: https://seismo-learn.org/sitepreview/seismo-learn/software/TauP-tutorial/taup/

Two thoughts/questions:

1. Should we move the software metadata to the [index page](https://seismo-learn.org/sitepreview/seismo-learn/software/TauP-tutorial/taup/)?
2. Do we really need to show the TOC in the index page?

File: source/taup/intro.rst [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+====
+
+:软件名称: TauP
+:主页: https://www.seis.sc.edu/taup/
+:源码地址: https://github.com/crotwell/TauP
+:作者: `Philip Crotwell <https://sc.edu/study/colleges_schools/artsandsciences/earth_ocean_and_environment/our_people/directory/crotwell_philip.php>`__
+:最新版本: v2.4.5 (2017-11-03)
```
Comment (seisman) [inline source/taup/intro.rst:8]:
```suggestion
:最新版本: v2.4.5 (2017-11-02)
```

File: source/taup/intro.rst [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+:适用平台: Linux、macOS、Windows
+:编程语言: Java
+:许可协议: GPL
+
+:教程作者: `seismo-learn <https://seismo-learn.org>`__
```
Comment (seisman) [inline source/taup/intro.rst:13]: Perhaps we should list the real author name (you), not the virtual "seismo-learn"?

File: source/taup/intro.rst [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+:教程更新日期: 2020-12-28
```
Comment (seisman) [inline source/taup/intro.rst:14]: I think we need to update the date if we add more things later, right?

File: source/taup/intro.rst [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+.. note::
+
+   本教程仅在 Linux 平台下使用 v2.4.5 测试通过，其余版本或平台可能有所差异。
```
Comment (seisman) [inline source/taup/intro.rst:18]: Are you working on Linux or macOS?

File: source/taup/intro.rst [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+--------------------------------------------------------------------------
+
+TauP 是一款常用于计算地震波走时和路径的 Java 程序。其原理与 ttimes 一样，基于 `Buland & Chapman (1983) <https://pubs.geoscienceworld.org/ssa/bssa/article-abstract/73/5/1271/118430/The-computation-of-seismic-travel-times?redirectedFrom=fulltext>`_ 提出的方法。
```
Comment (seisman) [inline source/taup/intro.rst:22]:
```suggestion
TauP 是一款常用于计算地震波走时和路径的 Java 程序。
```

Please add the references at the end of this section.

**Comment by core-man** [comment]
good suggestion. It seems good to hide toc in index since we can see them in side bar. in  this way, we should move metadata to index

File: source/taup/index.rst [hunk 12]
```diff
@@ -0,0 +1,7 @@
+TauP
+====
+
+.. toctree::
+
+   intro
```
Comment (seisman) [inline source/taup/index.rst:6]: introduction or intro?

File: source/taup/install.rst [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+
+    # 在 CentOS 7 上
+    $ sudo yum install java
+
+    # 在 Ubuntu 上
+    $ sudo apt update
+    $ sudo apt install default-jre default-jdk
+
+    # 在 macOS 上
+    $ brew install openjdk
```
Comment (seisman) [inline source/taup/install.rst:23]: Need to indent the whole block by 4 whitespaces.

File: source/taup/intro.rst (hunk 10 above)
Comment (core-man) [reply source/taup/intro.rst:18]: centos 7

Comment (seisman) [inline source/taup/intro.rst:18]:
```suggestion
   本教程仅在 Linux 和 macOS 平台下使用 v2.4.5 测试通过，其余版本或平台可能有所差异。
```

Comment (seisman) [reply source/taup/intro.rst:18]: I'll test it on macOS.

File: source/taup/index.rst (hunk 12 above)
Comment (core-man) [reply source/taup/index.rst:6]: let's use a word instead of shortcut?

Comment (seisman) [reply source/taup/index.rst:6]: Yes to me

File: source/taup/intro.rst (hunk 8 above)
Comment (core-man) [reply source/taup/intro.rst:13]: if true name is used, either nickname or real name? meanwhile, I would suggest to list all the authors in the following order, including original author, coding author and review author

File: source/taup/index.rst (hunk 12 above)
Comment (core-man) [reply source/taup/index.rst:6]: okay. make it as a rule in #27 ?

File: source/taup/intro.rst (hunk 8 above)
Comment (seisman) [reply source/taup/intro.rst:13]: > if true name is used, either nickname or real name? 

Either is ok.

Comment (seisman) [reply source/taup/intro.rst:13]: > I would suggest to list all the authors in the following order, including original author, coding author and review author

Looks good to me.

File: source/taup/index.rst (hunk 12 above)
Comment (seisman) [reply source/taup/index.rst:6]: OK. you can directly push to that branch.

File: source/taup/intro.rst (hunk 8 above)
Comment (core-man) [reply source/taup/intro.rst:13]: do you think we should explicitly indicate the role of those authors behind their names? or just a list of names?

File: source/taup/intro.rst (hunk 9 above)
Comment (core-man) [reply source/taup/intro.rst:14]: okay~

File: source/taup/index.rst (hunk 12 above)
Comment (seisman) [reply source/taup/index.rst:6]: Done.

Comment (seisman) [inline source/taup/index.rst:6]:
```suggestion
   introduction
```

File: source/taup/intro.rst [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+2. 输出的格式不够灵活，有时需要自己写脚本从输出中提取信息
+
+工具
+----
+
+TauP 提供了多个工具:
+
+- 图形界面： ``taup``
+- 交互式： ``taup_console``
+- 命令行工具： ``taup_time`` 、 ``taup_pierce`` 、 ``taup_setsac`` 、 ``taup_path`` 、 ``taup_curve`` 、 ``taup_create`` 、 ``taup_table`` 、 ``taup_wavefront``
```
Comment (seisman) [inline source/taup/intro.rst:49]: How to cite the tool:
```
Crotwell, H. P., T. J. Owens, and J. Ritsema (1999). The TauP Toolkit: Flexible seismic travel-time and ray-path utilities, Seismological Research Letters 70, 154–160.
```

File: source/taup/install.rst [hunk 15]
```diff
@@ -0,0 +1,47 @@
...
+否则需要先安装 Java 运行环境::
+
+    # 在 CentOS 7 上
+    $ sudo yum install java
+
+    # 在 Ubuntu 上
+    $ sudo apt update
+    $ sudo apt install default-jre default-jdk
+
+    # 在 macOS 上
```
Comment (seisman) [inline source/taup/install.rst:22]:
```suggestion
    # macOS homebrew用户
```

File: source/taup/install.rst [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+
+    $ java -version
+    openjdk version "1.8.0_161"
+    OpenJDK Runtime Environment (build 1.8.0_161-b14)
+    OpenJDK 64-Bit Server VM (build 25.161-b14, mixed mode)
+
+否则需要先安装 Java 运行环境::
+
+    # 在 CentOS 7 上
+    $ sudo yum install java
```
Comment (seisman) [inline source/taup/install.rst:16]:
```suggestion
    # CentOS/RHEL
    $ sudo yum install java
    
    # Fedora
    $ sudo dnf install java-latest-openjdk
```

File: source/taup/install.rst [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+
+    # 在 Ubuntu 上
+    $ sudo apt update
+    $ sudo apt install default-jre default-jdk
```
Comment (seisman) [inline source/taup/install.rst:20]:
```suggestion
    # Ubuntu/Debian
    $ sudo apt update
    $ sudo apt install default-jdk
```

File: source/taup/intro.rst (hunk 8 above)
Comment (seisman) [reply source/taup/intro.rst:13]: Yes, it sounds a good idea.

Perhaps we can list the different roles:

- 教程作者
- 教程贡献者
- 教程审稿人

File: source/taup/intro.rst [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+3. 内置震相解析器，支持不常见的震相名
+4. 自带多个预定义地球模型：``iasp91``、``prem``、``ak135``、``jb``、``1066a``、``1066b``、``pwdk``、``sp6``、``herrin``
+5. 部分兼容 ttimes，比如用 ttp 表示常见 P 波震相，除此之外，还有 tts、ttp+、tts+、ttbasic、ttall
+6. 采用线性插值，可能带来 0.01 秒的误差
+7. 直接在球坐标系下求解方程，不必做展平变换；与部分震相的解析解相比，最大误差为 0.01 秒，而 ttimes 的最大误差为 0.05 秒
+
+目前来看，有一些缺点：
+
+1. 不支持 PKPab、PKPbc 等震相名
+2. 输出的格式不够灵活，有时需要自己写脚本从输出中提取信息
```
Comment (seisman) [inline source/taup/intro.rst:40]: Need to rephase these sentences.

File: source/taup/intro.rst (hunk 8 above)
Comment (seisman) [reply source/taup/intro.rst:13]: @core-man I just updated the template following the above discussions. Please see if you agree with the latest template.

File: source/taup/install.rst (hunk 17 above)
Comment (core-man) [reply source/taup/install.rst:23]: How do you test different enviroments?

Comment (seisman) [reply source/taup/install.rst:23]: I can't test it. It's based on my understanding of JDK and the search results from https://pkgs.org/, so they may work or not.

File: source/taup/install.rst (hunk 13 above)
Comment (core-man) [reply source/taup/install.rst]: It seems error if we don't indent the block by whitespaces.

Comment (seisman) [reply source/taup/install.rst]: Maybe

File: source/taup/intro.rst (hunk 18 above)
Comment (core-man) [reply source/taup/intro.rst]: TauP features and tools rephrased. Please comment below

File: source/taup/introduction.rst [hunk 19]
```diff
@@ -0,0 +1,48 @@
+简介
+====
+
+TauP 用于计算一维球状地球速度模型下，地震震相的走时和路径。
```
Comment (seisman) [inline source/taup/introduction.rst:4]:
```suggestion
TauP 用于计算一维球状分层模型下地震震相的走时和路径。
```

File: source/taup/introduction.rst [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+
+功能与特色
+----------
+
+与传统的地震震相走时计算软件 ttimes 相比，TauP 具有更多的功能和特色：
+
+1. 计算震相走时、射线参数、反射点、穿透点、射线路径、走时曲线等
+2. 内置震相解析器，支持任意震相名
+3. 部分兼容 ttimes 震相名： ``ttp`` 表示常见 P 波震相；同理，还有 ``tts``、``ttp+``、``tts+``、``ttbasic``、``ttall``
+4. 自带多个地球地震波速度参考模型：``iasp91``、``prem``、``ak135``、``jb``、``1066a``、``1066b``、``pwdk``、``sp6``、``herrin`` 等
```
Comment (seisman) [inline source/taup/introduction.rst:14]: No need to list all models here
```suggestion
4. 自带多个地球地震波速度参考模型：iasp91、prem、ak135 等
```

File: source/taup/introduction.rst [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+5. 支持自定义速度模型
+6. 直接在球坐标系下求解方程，不必做展平变换，与部分震相的解析解相比，TauP 的最大误差为 0.01 秒，而 ttimes 的最大误差为 0.05 秒
+7. ObsPy 提供了 `TauP 端口 <https://docs.obspy.org/packages/obspy.taup.html>`__
```
Comment (seisman) [inline source/taup/introduction.rst:17]:
```suggestion
7. ObsPy 提供了 `TauP 的 Python 接口 <https://docs.obspy.org/packages/obspy.taup.html>`__
```

File: source/taup/install.rst [hunk 22]
```diff
@@ -0,0 +1,50 @@
...
+
+    # Ubuntu/Debian
+    $ sudo apt update
+    $ sudo apt install default-jdk
+
+    # macOS homebrew 用户
+    $ brew install openjdk
+
+TauP 最新版本为 2.4.5，该版本要求 Java 版本大于 1.8。若你的机器的 Java 版本小于 1.8，
+可以考虑下载使用旧版本的 TauP。但要注意，旧版本 TauP 可能存在已知的 BUG。
```
Comment (seisman) [inline source/taup/install.rst:29]: Need indentation.

File: source/taup/install.rst [hunk 23]
```diff
@@ -0,0 +1,50 @@
...
+
+    $ wget http://www.seis.sc.edu/downloads/TauP/TauP-2.4.5.tgz
+
+3. 解压::
+
+    $ tar -xvf TauP-2.4.5.tgz
+
+4. 安装::
+
+    $ sudo mv TauP-2.4.5 /opt
```
Comment (seisman) [inline source/taup/install.rst:41]: Safer to have a trailing slash
```suggestion
    $ sudo mv TauP-2.4.5 /opt/
```

File: source/taup/install.rst [hunk 24]
```diff
@@ -0,0 +1,50 @@
...
+
+5. 修改环境变量::
```
Comment (seisman) [inline source/taup/install.rst:43]:
```suggestion
5. 修改环境变量，将 TauP 的 bin 路径添加到 PATH 中::
```

File: source/taup/install.rst [hunk 25]
```diff
@@ -0,0 +1,50 @@
...
+
+    $ echo 'export PATH=/opt/TauP-2.4.5/bin:${PATH}' >> ~/.bashrc
+    $ source ~/.bashrc
+
+6. 执行测试
+
+终端键入 ``taup`` 若出现 TauP 图形界面则表示安装成功。
```
Comment (seisman) [inline source/taup/install.rst:50]:
```suggestion
6. 检查安装是否成功

    终端键入 ``taup`` 若出现 TauP 图形界面则表示安装成功。
```

File: source/taup/install.rst (hunk 22 above)
Comment (seisman) [reply source/taup/install.rst:29]: If you use VScode, you can select the whole block and press the `Tab` key to indent.

Comment (core-man) [reply source/taup/install.rst:29]: If I indentate them, it looks not good.

indentation

Comment (seisman) [reply source/taup/install.rst]: OK. I did the indentation for you.

**Review by seisman — APPROVED** [review]
Looks good to me. 🚀

**Comment by core-man** [comment]
Okay. Let's move on and come back when some improvements are needed in the future.
